parser/src/lib.rs

```diff
@@ -44,20 +52,39 @@ pub enum DisplayHint {
     Unknown(String),
 }
 
-impl FromStr for DisplayHint {
-    type Err = ();
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        Ok(match s {
-            "µs" => DisplayHint::Microseconds,
-            "a" => DisplayHint::Ascii,
-            "b" => DisplayHint::Binary,
-            "x" => DisplayHint::Hexadecimal { uppercase: false },
-            "X" => DisplayHint::Hexadecimal { uppercase: true },
-            "?" => DisplayHint::Debug,
-            _ => return Err(()),
-        })
-    }
+fn parse_display_hint(mut s: &str) -> Option<DisplayHint> {
```

Comment:
Because it's an internal implementation detail, I don't see a benefit to using `FromStr`, because a) it has the Err type parameter that we don't need or use, and it means that we special case parsing the display hint compared to other stuff. Happy to hear other people's opinions.

Personally I think the best format for these functions is to follow the `nom` pattern, where you have functions like `fn parse_*(input: &[u8]) -> Option<(&[u8], Type)>`, where the unconsumed input is returned along with the parsed value. This makes it easy to chain parsing functions together. (The DSL here is simple enough that these are not things that need much thought IMO).